gst/nnstreamer/tensor_sparse/tensor_sparse_util.c

```diff
@@ -0,0 +1,207 @@
+
+  /** alloc maximum possible size of memory */
+  values = g_malloc0 (element_size * element_count);
+  indices = g_malloc0 (sizeof (guint) * element_count);
+
+  data_type = meta->type;
+
+  /** Consider using macro to reduce loc and readability */
+  for (i = 0; i < element_count; ++i) {
+    switch (data_type) {
+      case _NNS_INT32:
+        if (((int32_t *) in)[i] != 0) {
+          ((int32_t *) values)[nnz] = ((int32_t *) in)[i];
+          indices[nnz] = i;
+          nnz += 1;
+        }
+        break;
+      case _NNS_UINT32:
+        if (((uint32_t *) in)[i] != 0) {
+          ((uint32_t *) values)[nnz] = ((uint32_t *) in)[i];
+          indices[nnz] = i;
+          nnz += 1;
+        }
+        break;
+      case _NNS_INT16:
+        if (((int16_t *) in)[i] != 0) {
+          ((int16_t *) values)[nnz] = ((int16_t *) in)[i];
+          indices[nnz] = i;
+          nnz += 1;
+        }
+        break;
+      case _NNS_UINT16:
+        if (((uint16_t *) in)[i] != 0) {
+          ((uint16_t *) values)[nnz] = ((uint16_t *) in)[i];
+          indices[nnz] = i;
+          nnz += 1;
+        }
+        break;
+      case _NNS_INT8:
+        if (((int8_t *) in)[i] != 0) {
+          ((int8_t *) values)[nnz] = ((int8_t *) in)[i];
+          indices[nnz] = i;
+          nnz += 1;
+        }
+        break;
+      case _NNS_UINT8:
+        if (((uint8_t *) in)[i] != 0) {
+          ((uint8_t *) values)[nnz] = ((uint8_t *) in)[i];
+          indices[nnz] = i;
+          nnz += 1;
+        }
+        break;
+      case _NNS_FLOAT64:
+        if (((double *) in)[i] != 0) {
+          ((double *) values)[nnz] = ((double *) in)[i];
+          indices[nnz] = i;
+          nnz += 1;
+        }
+        break;
+      case _NNS_FLOAT32:
+        if (((float *) in)[i] != 0) {
+          ((float *) values)[nnz] = ((float *) in)[i];
+          indices[nnz] = i;
+          nnz += 1;
+        }
+        break;
+      case _NNS_INT64:
+        if (((int64_t *) in)[i] != 0) {
+          ((int64_t *) values)[nnz] = ((int64_t *) in)[i];
+          indices[nnz] = i;
+          nnz += 1;
+        }
+        break;
+      case _NNS_UINT64:
+        if (((uint64_t *) in)[i] != 0) {
+          ((uint64_t *) values)[nnz] = ((uint64_t *) in)[i];
+          indices[nnz] = i;
+          nnz += 1;
+        }
+        break;
+      default:
+        nns_loge ("Error occured during get tensor value");
+        return NULL;
+    }
+  }
+
+  /** update meta nnz info */
+  meta->sparse_info.nnz = nnz;
+
+  /** write to out */
+  output_size = element_size * nnz + sizeof (guint) * nnz;
+  output = g_malloc0 (output_size);
+  memcpy (output, values, element_size * nnz);
+  memcpy ((guint8 *) output + (element_size * nnz),
+      indices, sizeof (guint) * nnz);
+
+  g_free (values);
+  g_free (indices);
+
+  return gst_tensor_meta_info_append_header (meta,
+      gst_memory_new_wrapped (0, output, output_size, 0,
```

Comment:
mem leak, gst_tensor_meta_info_append_header() does not release memory.

recommend to alloc data + header and wrap.
```
output_size += header_size (from meta);
output = g_malloc0 (output_size);
gst_tensor_meta_info_update_header (meta, output); <<- this will set header
```